src/main/java/org/apache/sysds/runtime/matrix/data/LibCommonsMath.java

```diff
@@ -252,6 +295,104 @@ private static EigenDecomposition computeEigenRegularized(MatrixBlock in) {
 			DataConverter.convertToArray2DRowRealMatrix(in2));
 	}
 
+	/**
+	 * Function to perform FFT on a given matrix.
+	 *
+	 * @param re matrix object
+	 * @return array of matrix blocks
+	 */
+	private static MatrixBlock[] computeFFT(MatrixBlock re) {
+		if(re == null)
+			throw new DMLRuntimeException("Invalid empty block");
+		if (re.isEmptyBlock(false)) {
+			// Return the original matrix as the result
+			return new MatrixBlock[]{re, new MatrixBlock(re.getNumRows(), re.getNumColumns(), true)}; // Assuming you need to return two matrices: the real part and an imaginary part initialized to 0.
+		}
+		// run fft
+		re.sparseToDense();
+		return fft(re);
+	}
+
+	private static boolean isMatrixAllZeros(MatrixBlock matrix) {
```

Comment:
This is not a good check, please use isEmpty() and nonzeros count to verify if there is values in the matrix.

Reply:
was removed since it was not even used (possibly a leftover of the handler for zero matrix inputs, which was removed again, as it will be introduced in the PythonTest PR).